Stm32h755 soc support and nucleo-h755zi-q support

mbanducci-sandc opened the pull request:
This PR adds support for the stm32h755 soc and the ST nucleo-h755zi-q development board. 

The h755 is a close relative of the h745 which adds cryptography and hash accelerator peripherals.

The nucleo-h755zi-q is a nucleo-144 development board for the stm32h755. It shares the same layout as other, similar nucleo-144 boards such as the nucleo-h745zi-q already supported by Zephyr.

Thanks!

mbanducci-sandc committed:
soc: stm32: Add support for the stm32h755

Add support for the stm32h755 which is a close relative of
the stm32h745 with additional cryptography and hashing
peripherals.

Signed-off-by: Mike Banducci <michael.banducci@sandc.com>

mbanducci-sandc committed:
boards: stm32: Add nucleo_h755zi_q support

Add support for the ST nucleo h755zi_q board which
uses an stm32h755 in the st nucleo-144 board
format.

Signed-off-by: Mike Banducci <michael.banducci@sandc.com>

ajarmouni-st commented on boards/st/nucleo_h755zi_q/doc/index.rst:
```diff
@@ -0,0 +1,293 @@
+========
+
+Nucleo H755ZI-Q board includes an ST-LINK/V3 embedded debug tool interface.
+
```
```

The board is configured to be flashed using west `STM32CubeProgrammer`_ runner
for both cores, so its installation is required to be able to flash the board.
The target core is detected automatically.

It is advised to use `STM32CubeProgrammer`_ to check and update option bytes
configuration and flash ``nucleo_h755zi_q/stm32h755xx/m7`` and
``nucleo_h755zi_q/stm32h755xx/m4`` targets.

```

ajarmouni-st commented on boards/st/nucleo_h755zi_q/doc/index.rst:
```diff
@@ -0,0 +1,293 @@
+
+Flashing operation will depend on the target to be flashed and the SoC
+option bytes configuration.
+
```
```

It is advised to use `STM32CubeProgrammer`_ to check and update option bytes
configuration.

```

ajarmouni-st commented on boards/st/nucleo_h755zi_q/doc/index.rst:
```diff
@@ -0,0 +1,293 @@
+board is powered (Option bytes BCM7 and BCM4 are checked).
+In that configuration, Kconfig boot option ``STM32H7_BOOT_CM4_CM7`` should be selected.
+Zephyr flash configuration has been set to meet these default settings.
+
```
```

Alternatively, openocd or JLink can also be used to flash the board using
the ``--runner`` (or ``-r``) option:

.. code-block:: console

   $ west flash --runner openocd
   $ west flash --runner jlink
```

ajarmouni-st commented on boards/st/nucleo_h755zi_q/doc/index.rst:
```diff
@@ -0,0 +1,293 @@
+   https://mbd.kleier.net/integrating-st-link-v3.html
+
+.. _STM32CubeIDE:
+   https://www.st.com/en/development-tools/stm32cubeide.html
```
```

.. _STM32CubeProgrammer:
   https://www.st.com/en/development-tools/stm32cubeprog.html
```

ajarmouni-st commented on boards/st/nucleo_h755zi_q/nucleo_h755zi_q.dtsi:
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
```
copyright

ajarmouni-st commented on boards/st/nucleo_h755zi_q/nucleo_h755zi_q_stm32h755xx_m4_defconfig:
```diff
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2020 Alexander Kozhinov <AlexanderKozhinov@yandex.com>
```
copyright

ajarmouni-st commented on boards/st/nucleo_h755zi_q/nucleo_h755zi_q_stm32h755xx_m7.dts:
```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2020 Alexander Kozhinov <AlexanderKozhinov@yandex.com>
```
copyright

ajarmouni-st commented on boards/st/nucleo_h755zi_q/nucleo_h755zi_q_stm32h755xx_m7_defconfig:
```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2020 Alexander Kozhinov <AlexanderKozhinov@yandex.com>
```
copyright

ajarmouni-st commented on boards/st/nucleo_h755zi_q/support/openocd.cfg:
```diff
@@ -0,0 +1,30 @@
+# STM32H745ZI Nucleo board OpenOCD ST-LINK V3 configuration
+#
+# Copyright (c) 2020 Alexander Kozhinov  <AlexanderKozhinov@yandex.com>
```
copyright

ajarmouni-st commented on samples/drivers/flash_shell/sample.yaml:
should be in a different commit

ajarmouni-st reviewed (dismissed):
Thanks for working on this, here is a first batch of comments.

mbanducci-sandc committed:
samples: flash: Exclude the stm32h755xx/m4 from flash samples

Exclude the nucleo_h755i_q m4 core from the flash_shell samples,
as the m4 core in stm32h7 parts does not have flash support in Zephyr.

Signed-off-by: Mike Banducci <michael.banducci@sandc.com>

nordicjm commented on boards/st/nucleo_h755zi_q/board.cmake:
```diff
@@ -0,0 +1,13 @@
+board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
+board_runner_args(jlink "--device=STM32H755ZI" "--speed=3300")
+if(CONFIG_BOARD_NUCLEO_H755ZI_Q_STM32H755XX_M7)
+board_runner_args(openocd --target-handle=_CHIPNAME.cpu0)
```
2 space indent

nordicjm commented on boards/st/nucleo_h755zi_q/board.cmake:
```diff
@@ -0,0 +1,13 @@
+if(CONFIG_BOARD_NUCLEO_H755ZI_Q_STM32H755XX_M7)
+board_runner_args(openocd --target-handle=_CHIPNAME.cpu0)
+elseif(CONFIG_BOARD_NUCLEO_H755ZI_Q_STM32H755XX_M4)
+board_runner_args(openocd --target-handle=_CHIPNAME.cpu1)
```
and here

nordicjm commented on boards/st/nucleo_h755zi_q/doc/img/nucleo_h755zi_q.jpg:
convert to webp, then put through https://tinypng.com/

nordicjm commented on boards/st/nucleo_h755zi_q/doc/index.rst:
```diff
@@ -0,0 +1,310 @@
+
+It is advised to use `STM32CubeProgrammer`_ to check and update option bytes
+configuration and flash ``nucleo_h755zi_q/stm32h755xx/m7`` and
+``nucleo_h755zi_q/stm32h755xx/m4`` targets.
```
*board targets

nordicjm commented on boards/st/nucleo_h755zi_q/nucleo_h755zi_q_stm32h755xx_m4_defconfig:
```diff
@@ -0,0 +1,20 @@
+#CONFIG_CONSOLE=y
+#CONFIG_UART_CONSOLE=y
+
+# enable pin controller
```
*Enable

nordicjm commented on boards/st/nucleo_h755zi_q/nucleo_h755zi_q_stm32h755xx_m7_defconfig:
```diff
@@ -0,0 +1,27 @@
+# Enable Clock
+CONFIG_CLOCK_CONTROL=y
+
+# enable pin controller
```
*Enable

nordicjm commented on boards/st/nucleo_h755zi_q/nucleo_h755zi_q_stm32h755xx_m7_defconfig:
```diff
@@ -0,0 +1,27 @@
+# Enable HW stack protection
+CONFIG_HW_STACK_PROTECTION=y
+
+# Enable UART ( disable to assign to M4 core)
+CONFIG_SERIAL=y
+
+# Console ( disable to assign to M4 core)
```
remove space after (

mkaricheri commented:
Would be great if we can see few more approval for this pull request. Thanks

FRASTM commented on boards/st/nucleo_h755zi_q/doc/index.rst:
```diff
@@ -0,0 +1,295 @@
+
+By default:
+
+  - CPU0 (Cortex-M7) boot address is set to 0x80000000 (OB: BOOT_CM7_ADD0)
```
```suggestion
  - CPU0 (Cortex-M7) boot address is set to 0x08000000 (OB: BOOT_CM7_ADD0)
```
(this is also wrong for the stm32h747i_disco)

FRASTM commented on boards/st/nucleo_h755zi_q/doc/index.rst:
```diff
@@ -0,0 +1,295 @@
+By default:
+
+  - CPU0 (Cortex-M7) boot address is set to 0x80000000 (OB: BOOT_CM7_ADD0)
+  - CPU1 (Cortex-M4) boot address is set to 0x81000000 (OB: BOOT_CM4_ADD0)
```
```suggestion
  - CPU1 (Cortex-M4) boot address is set to 0x08100000 (OB: BOOT_CM4_ADD0)
```
(this is also wrong for the stm32h747i_disco)

FRASTM reviewed (commented):
LGTM

mkaricheri reviewed (approved):
Looks good to me

mkaricheri commented:
@erwango we are Curious as to why this is still waiting to be merged? Is there anything to be done at our end? Thanks
